### PR TITLE
Adds support for Color Sensor

### PR DESCRIPTION
**Pull request by mdesjardins**
* Adds REV Robotics ColorSensor3 vendor deps
* Adds rudimentary subsystem for the spinner

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -57,4 +57,3 @@ run/
 gradle/.launch/
 !gradle/*/*.jar
 */Version.java
-src/main/java/org/frc5687/infiniterecharge/robot/util/Version.java
```
Comment (BenBernardCIS) [inline .gitignore]: Doesn't this effectively add Version.java back to version control?

File: src/main/java/org/frc5687/infiniterecharge/robot/commands/DriveSpinner.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @Override
+    public void execute() {
+        /**
+         * Execute is where the code will be running continuously until isFinished() has returned true
+         * execute() is run every 20ms(unless told otherwise)
+         * this is where closed loop and open loop control is taken place.
+         */
+        super.execute();
```
Comment (BenBernardCIS) [inline src/main/java/org/frc5687/infiniterecharge/robot/commands/DriveSpinner.java:40]: The excecute() method needs to send motor outputs.

File: src/main/java/org/frc5687/infiniterecharge/robot/subsytems/Spinner.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        /**
+         * Inside the constructor is where we instantiate our objects and variable.
+         * Different objects requires different parameters, documentation of the object provides the order and what parameters are needed.
+         * Ex.
+         * _spinnerSpark = new CANSparkMax(CAN ID, TYPE OF MOTOR);
+         * CAN ID's are placed in RobotMap so replace with RobotMap.CAN.SPARKMAX.SPINNER
+         * TYPE OF MOTOR: we use brushless so CANSparkMaxLowLevel.MotorType.kBrushless <--- this is provided my RevRobotics vendor library.
+         */
+
+        // TODO(mike) do we need to move this to RobotMap? I think we only have one i2c bus?
```
Comment (BenBernardCIS) [inline src/main/java/org/frc5687/infiniterecharge/robot/subsytems/Spinner.java]: We should either move to Constants or just skip the local variable.  Not critical though.

File: src/main/java/org/frc5687/infiniterecharge/robot/subsytems/Spinner.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    @Override
+    public void periodic() {
+        /**
+         * Use the space to create what command the Subsystem will be running periodically (20ms)
+         * Ex.
+         * setDefaultCommand(new DriveSpinner(parameters in here));
+         */
+        setDefaultCommand(new DriveSpinner(this));
```
Comment (BenBernardCIS) [inline src/main/java/org/frc5687/infiniterecharge/robot/subsytems/Spinner.java]: This shouldn't be happening in periodic()....that will create a new DriveSpinner command every loop through.
Not sure why the sample shows that!  This should move to the constructor.

File: src/main/java/org/frc5687/infiniterecharge/robot/commands/DriveSpinner.java (hunk 2 above)
Comment (BenBernardCIS) [reply src/main/java/org/frc5687/infiniterecharge/robot/commands/DriveSpinner.java:40]: I realize that this is template code and not intended to be complete...so you don't need to change it now.

**Review by deslobodzian — APPROVED** [review]
Looks all good

**Review by BenBernardCIS — APPROVED** [review]

**Pull request merged**